.github/actions/build-test/windows/action.yml

```diff
@@ -0,0 +1,168 @@
+  BUILD_TYPE:
+    required: true
+  SHARED_LIBS_TOGGLE:
+    required: true
+  OPENSSL_TOGGLE:
+    required: true
+  ARCH_CHOCO_OPTIONS:
+    required: true
+  ARCH_ADDRESS_MODEL:
+    required: true
+  ARCH_CMAKE:
+    required: true
+  INSTALL_BOOST:
+    required: true
+  INSTALL_THRIFT:
+    required: true
+  BOOST_URL:
+    required: true
+  BOOST_ARCHIVE_NAME:
+    required: true
+  BOOST_FOLDER_NAME:
+    required: true
+  BOOST_INCLUDE_FOLDER:
+    required: true
+  RUN_TESTS:
+    required: true
+  HAZELCAST_ENTERPRISE_KEY:
+    required: true
+  AWS_ACCESS_KEY_ID:
+    required: true
+  AWS_SECRET_ACCESS_KEY:
+    required: true
+  HZ_TEST_AWS_INSTANCE_PRIVATE_IP:
+    required: true
+
+env:
+  # Not possible to set this as a default
+  # https://github.com/orgs/community/discussions/46670
+  shell: pwsh
+
+runs:
+  using: composite
+  steps:
+    - name: Read Config
+      shell: bash
+      run: cat .github/config.env >> $GITHUB_ENV
+
+    - name: Download hazelcast-enterprise-tests.jar
+      shell: ${{ env.shell }}
+      run: |
+        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+        Invoke-WebRequest https://raw.githubusercontent.com/hazelcast/private-test-artifacts/data/certs.jar -Headers @{"Authorization"="token ${{ inputs.GH_TOKEN }}"} -OutFile hazelcast-enterprise-${{ env.HAZELCAST_VERSION }}-tests.jar
+
+    - name: Install SysInternals
+      shell: ${{ env.shell }}
+      run: |
+          Invoke-WebRequest `
+              "https://download.sysinternals.com/files/SysinternalsSuite.zip" `
+              -OutFile "sys-internals.zip" `
+              -UserAgent "''"
+          Expand-Archive sys-internals.zip -DestinationPath sys-internals
+          ls sys-internals
+
+    - uses: nick-fields/retry@v2
+      name: Install OpenSSL
+      if: inputs.OPENSSL_TOGGLE == 'ON'
+      with:
+        timeout_minutes: 10
+        max_attempts : 4
+        retry_on: error
+        shell: ${{ env.shell }}
+        command: |
+          choco install openssl ${{ inputs.ARCH_CHOCO_OPTIONS }}
+          Invoke-WebRequest `
+              "https://curl.se/ca/cacert.pem" `
+              -OutFile "C:\cacert.pem"
+
+    - if: ${{ inputs.INSTALL_BOOST != 'false' }}
+      name: Install Boost
+      shell: ${{ env.shell }}
+      run: |
+        Invoke-WebRequest `
+            "${{ inputs.BOOST_URL }}" `
+            -OutFile "${{ inputs.BOOST_ARCHIVE_NAME }}" `
+            -UserAgent "''"
+        tar xzf ${{ inputs.BOOST_ARCHIVE_NAME }}
+        rm ${{ inputs.BOOST_ARCHIVE_NAME }}
+        cd ${{ inputs.BOOST_FOLDER_NAME }}
+        .\bootstrap.bat
+        .\b2 address-model=${{ inputs.ARCH_ADDRESS_MODEL }} --with-atomic --with-thread --with-chrono install
+        cd ..
+        Remove-Item ${{ inputs.BOOST_FOLDER_NAME }} -Recurse -Force
+
+    - if: ${{ inputs.INSTALL_THRIFT != 'false' }}
+      name: Install Thrift
+      shell: ${{ env.shell }}
+      run: |
+        Invoke-WebRequest `
```

Comment:
move to ps1

Reply:
https://github.com/hazelcast/hazelcast-cpp-client/pull/1243#discussion_r1764850057